### PR TITLE
feat(secret): Support for detecting Hugging Face Access Tokens

### PR DESCRIPTION
**Pull request by chrisking**
## Description

Hugging Face Access Tokens are not detected by the scanner for secrets yet, I added them as an entity with this change and provided a test case as well.

## Related issues
- Closes https://github.com/aquasecurity/trivy/issues/6262

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [x] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/fanal/secret/builtin-rules.go [hunk 1]
```diff
@@ -158,6 +158,15 @@ var builtinRules = []Rule{
 		Regex:    MustCompile(`glpat-[0-9a-zA-Z\-\_]{20}`),
 		Keywords: []string{"glpat-"},
 	},
+	{
+	        ID:              "hugging-face-access-token",
+	        Category:        types.SecretRuleCategory("HuggingFace"),
+	        Severity:        "CRITICAL",
+	        Title:           "Hugging Face Access Token",
+	        Regex:           MustCompile(`hf_[A-Za-z0-9]{39}`),
+	        SecretGroupName: "secret",
```
Comment (DmitriyLewen) [inline pkg/fanal/secret/builtin-rules.go:167]: You don't use groups in regex.
This field is needed to take/hide part of the found string.
If I understand correctly, you need to hide the full line, so you can delete it.

File: pkg/fanal/secret/scanner_test.go [hunk 2]
```diff
@@ -609,6 +609,28 @@ func TestSecretScanner(t *testing.T) {
 			},
 		},
 	}
+	wantFindingHF := types.SecretFinding{
```
Comment (DmitriyLewen) [inline pkg/fanal/secret/scanner_test.go:612]: You need to create a new test file + test case and use `wantFindingHF` as want result.

File: pkg/fanal/secret/builtin-rules.go [hunk 3]
```diff
@@ -158,6 +158,15 @@ var builtinRules = []Rule{
 		Regex:    MustCompile(`glpat-[0-9a-zA-Z\-\_]{20}`),
 		Keywords: []string{"glpat-"},
 	},
+	{
+	        ID:              "hugging-face-access-token",
+	        Category:        types.SecretRuleCategory("HuggingFace"),
+	        Severity:        "CRITICAL",
+	        Title:           "Hugging Face Access Token",
+	        Regex:           MustCompile(`hf_[A-Za-z0-9]{39}`),
```
Comment (DmitriyLewen) [inline pkg/fanal/secret/builtin-rules.go:166]: Can we add any link to documents with this secret format?

**Review by DmitriyLewen — COMMENTED** [review]
Hello @chrisking 
Thnka for your work!

I left comments. Take a look, when you have time, please.

Regards, Dmitriy

File: pkg/fanal/secret/builtin-rules.go (hunk 3 above)
Comment (DmitriyLewen) [reply pkg/fanal/secret/builtin-rules.go:166]: e.g. https://huggingface.co/docs/hub/en/security-tokens

Comment (chrisking) [reply pkg/fanal/secret/builtin-rules.go]: When you say add links, where should they be added specifically?

File: pkg/fanal/secret/builtin-rules.go (hunk 1 above)
Comment (chrisking) [reply pkg/fanal/secret/builtin-rules.go]: What do you mean by not using groups? I admit I'm not really familiar with Go as a language and was trying to reverse engineer what to do based on similar components within the codebase.

Comment (DmitriyLewen) [reply pkg/fanal/secret/builtin-rules.go]: I meant named regex groups ([example](https://regex101.com/r/RBC6wb/1)) - https://www.regular-expressions.info/named.html

Comment (DmitriyLewen) [reply pkg/fanal/secret/builtin-rules.go]: To save your time - i did refactor a bit.
Can you take a look and confirm that it works correctly?

Comment (DmitriyLewen) [reply pkg/fanal/secret/builtin-rules.go]: Group has been removed in https://github.com/aquasecurity/trivy/pull/6236/commits/5bc92a101e9ce54b62ffac2fbe8fbc59f3c9e3c4

File: pkg/fanal/secret/builtin-rules.go (hunk 3 above)
Comment (DmitriyLewen) [reply pkg/fanal/secret/builtin-rules.go]: added in https://github.com/aquasecurity/trivy/pull/6236/commits/5bc92a101e9ce54b62ffac2fbe8fbc59f3c9e3c4

File: pkg/fanal/secret/scanner_test.go (hunk 2 above)
Comment (DmitriyLewen) [reply pkg/fanal/secret/scanner_test.go]: Did this in https://github.com/aquasecurity/trivy/pull/6236/commits/758c4f5441f082ec421b9d5140130a971bfdde55

**Comment by chrisking** [comment]
Thanks for the help @DmitriyLewen , I really appreciate the extra help to get this code up to the standards for Trivy. What's next for this PR?

**Review by DmitriyLewen — APPROVED** [review]
@knqyf263 take a look, when you have time.

**Pull request merged**